### PR TITLE
fix: remove time from date intervals

### PR DESCRIPTION
**Pull request by rephus**
Closes: #3472

### Description:

Before:
Screenshot from 2022-11-08 09-49-22

After:
Screenshot from 2022-11-08 09-58-43

image

Screenshot from 2022-11-08 09-59-32

### CODE REVIEW

File: packages/common/src/utils/timeFrames.ts [hunk 1]
```diff
@@ -280,13 +280,13 @@ export const timeFrameConfigs: Record<TimeFrames, TimeFrameConfig> = {
...
         getSql: getSqlForTruncatedDate,
-        getAxisMinInterval: () => null,
+        getAxisMinInterval: () => 86400000,
     },
     WEEK: {
         getLabel: () => 'Week',
         getDimensionType: () => DimensionType.DATE,
         getSql: getSqlForTruncatedDate,
-        getAxisMinInterval: () => null,
+        getAxisMinInterval: () => 86400000,
```
Comment (IrakliJani) [inline packages/common/src/utils/timeFrames.ts:289]: I believe this fixes the bug but why 🤔 can you add a comment in the code as well?

Comment (ZeRego) [reply packages/common/src/utils/timeFrames.ts:289]: I believe forcing the minimum interval to be 1 day will prevent echarts from showing nice timelines. Waiting for PR env to be ready to confirm this.

Comment (rephus) [reply packages/common/src/utils/timeFrames.ts:289]: yeah, the same applies to all other axis intervals, 86400000 means 1 day in milliseconds. @ZeRego originally implemented this, so he probably knows what kind of dark magic  this is :D

Comment (rephus) [reply packages/common/src/utils/timeFrames.ts:289]: https://github.com/lightdash/lightdash/pull/3346#discussion_r984590514

Comment (IrakliJani) [reply packages/common/src/utils/timeFrames.ts:289]: :chefkiss:  [](edited-with-Axolo)

Comment (ZeRego) [reply packages/common/src/utils/timeFrames.ts:289]: In production:
Screenshot 2022-11-08 at 13 45 34

This PR:
Screenshot 2022-11-08 at 13 39 20

The problem with setting this value is that we force the axis points and lose the nice labels for the months and days. 
This is not a problem when dealing with large time intervals like year and quarter but uglier when dealing with day and week.

Comment (rephus) [reply packages/common/src/utils/timeFrames.ts:289]: Nice catch, let me have a look [](edited-with-Axolo)

Comment (ZeRego) [reply packages/common/src/utils/timeFrames.ts]: @TuringLovesDeathMetal would this work for you? is there a problem if we still show the `ticks`?

File: packages/frontend/src/hooks/echarts/useEcharts.ts [hunk 2]
```diff
@@ -526,6 +527,18 @@ const getEchartAxis = ({
...
+            isDimension(field) &&
+            (field.timeInterval === TimeFrames.WEEK ||
+                field.timeInterval === TimeFrames.DAY)
+        ) {
+            // If timeInterval is WEEK or DAY, we don't show time on the axis
+            // We don't use axisMinInterval becuase that breaks the dynamic format on dates
+            axisConfig.axisLabel = {
+                formatter: {
+                    hour: '',
+                },
```
Comment (ZeRego) [inline packages/frontend/src/hooks/echarts/useEcharts.ts:540]: Didn't know about this echarts setting 😮  really nice 🕺 
What do you think if we had this configuration in the `timeFrameConfigs` as we have for `axisMinInterval`?

Comment (rephus) [reply packages/frontend/src/hooks/echarts/useEcharts.ts:540]: yeah, good idea, if we are happy about formatter I can make the change [](edited-with-Axolo)

Comment (ZeRego) [reply packages/frontend/src/hooks/echarts/useEcharts.ts:540]: I’m happy with the formatter :+1: [](edited-with-Axolo)

**Review by ZeRego — APPROVED** [review]

**Pull request merged**